cert-manager-istio-csr.yaml

```diff
@@ -19,13 +19,66 @@ pipeline:
       output: cmd
       packages: ./cmd
 
+subpackages:
+  - name: ${{package.name}}-compat
+    description: Compatibility package for cert-manager CSI driver
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.contextdir}}"/ko-app/
+          ln -sf ../usr/bin/cmd "${{targets.contextdir}}"/ko-app/cmd
+    test:
+      environment:
+        contents:
+          packages:
+            - ${{package.name}}
+      pipeline:
+        - uses: test/tw/symlink-check
+
 update:
   enabled: true
   github:
     identifier: cert-manager/istio-csr
     strip-prefix: v
 
 test:
+  environment:
+    contents:
+      packages:
+        - curl
+    environment:
+      METRICS_PORT: "9402"
+      READINESS_PORT: "6060"
   pipeline:
-    - runs: |
-        cmd --help
+    - uses: test/tw/help-check
+      with:
+        bins: cmd
+    - uses: test/kwok/cluster
+    - name: Install cert-manager CRDs
+      runs: |
+        set -euo pipefail
+        kubectl apply -f https://github.com/cert-manager/cert-manager/releases/latest/download/cert-manager.crds.yaml
+    - uses: test/daemon-check-output
+      with:
+        setup: |-
+          kubectl create namespace istio-system
+        start: |-
+          cmd \
+            --log-level=2 \
+            --metrics-port=${METRICS_PORT} \
+            --readiness-probe-port=${READINESS_PORT} \
+            --readiness-probe-path=/readyz \
+            --serving-address=0.0.0.0:${GRPC_PORT} \
+            --certificate-namespace=istio-system \
+            --trust-domain=cluster.local
+        timeout: 120
+        expected_output: |
+          starting server
+          Starting metrics server
+        post: |-
+          set -o pipefail
+
+          # Test readiness endpoint
+          curl -sf "http://localhost:${READINESS_PORT}/readyz"
```

Comment:
nit: running curl commands without wait-for-it guardrail might lead flakes, same without `--retry`